src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs

```diff
@@ -109,8 +110,19 @@ public void Start<TContext>(IHttpApplication<TContext> application)
 
                     if (!parsedAddress.Host.Equals("localhost", StringComparison.OrdinalIgnoreCase))
                     {
-                        _disposables.Push(engine.CreateServer(
-                            parsedAddress));
+                        try
+                        {
+                            _disposables.Push(engine.CreateServer(parsedAddress));
+                        }
+                        catch (AggregateException ex)
+                        {
+                            if ((ex.InnerException as UvException)?.StatusCode == Constants.EADDRINUSE)
+                            {
+                                throw new IOException($"Failed to bind to address {parsedAddress}: address already in use.", ex);
```

Comment:
It's technically a breaking change not to keep this a AggregateException/UvException, but AggregateExceptions are gross enough that I don't think anyone is catching that explicitly. I guess IOException is fine.